src/pharmpy/workflows/dispatchers/local_dask.py

```diff
@@ -12,41 +13,63 @@
 
 def run(workflow: Workflow[T]) -> T:
 
-    with TemporaryDirectory() as tempdirname:
-        with chdir(tempdirname):
-            dsk = workflow.as_dask_dict()
+    # NOTE We change to a new temporary directory so that all files generated
+    # by the workflow end-up in the same root directory. Each task of a
+    # workflow has the responsibility to avoid collisions on the file system
+    # (using UUIDs as filenames for instance). This also allows tasks to share
+    # files on the local file-system if needed. Since in the future we may want
+    # to run tasks on remote machines it might be a good idea to instead not
+    # assume tasks share the same working directory or even filesystem, and
+    # let them communicate through some other mean and create their own
+    # temporary directory if needed.
```

Comment:
I guess tasks would not even be able to use `chdir` because they could be run on threads and that would not be nice.

Reply:
Updated.